library/core/src/array/mod.rs

```diff
@@ -463,6 +528,18 @@ where
     unsafe { collect_into_array(iter).unwrap_unchecked() }
 }
 
+// Infallible version of `collect_into_array_rslt_unchecked`.
+unsafe fn collect_into_array_unchecked<I, const N: usize>(iter: &mut I) -> [I::Item; N]
+where
+    I: Iterator + TrustedLen,
+{
+    let mut map = iter.map(|el| Ok::<_, Infallible>(el));
+
+    // SAFETY: Valid array elements are covered by the fact that all passed values
+    // to `collect_into_array` are `Ok`.
+    unsafe { collect_into_array_rslt_unchecked(&mut map).unwrap_unchecked() }
```

Comment:
No need for `unwrap_unchecked` here, you can do https://github.com/c410-f3r/rust/pull/3